cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -89,12 +91,15 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         builder.put(entry.getKey(), jobDataMap.getString(entry.getKey()));
       }
 
+      ProgramId programId = new ProgramId(namespaceId, applicationId, programType, programName);
```

Comment:
should use `ProgramId programId = new ApplicationId(namespaceId, applicationId, appVersion).program(programType, programId)` to include appVersion